src/main/java/database/helper/Settings.java

```diff
@@ -14,19 +14,24 @@
 import org.slf4j.LoggerFactory;
 
 public class Settings {
+    static final String DATABASE_STORAGE_PATH_PROPERTY_NAME = "database.storage.path";
+    static final String DEFAULT_DATABASE_STORAGE_PATH = "database";
+    static final int DEFAULT_LIMIT_VALUE = 100;
+    private static final String LIMIT_PROPERTY_NAME = "limit";
+    private static final Logger LOG = LoggerFactory.getLogger(Settings.class);
     private String propertyFileName;
     private final Path defaultJsonDatabasePath;
-    private static final Logger LOG = LoggerFactory.getLogger(Settings.class);
-    static final String DATABASE_STORAGE_PATH = "database.storage.path";
-    static final String DEFAULT_DATABASE_STORAGE_PATH = "database";
+    private final Properties properties = new Properties();
 
     public Settings() throws CreationDatabaseException {
         defaultJsonDatabasePath = getFilePath(DEFAULT_DATABASE_STORAGE_PATH);
+        loadProperties();
     }
 
     public Settings(String propertyFileName) throws CreationDatabaseException {
         this();
         this.propertyFileName = propertyFileName;
+        loadProperties();
```

Comment:
You already load properties in default constructor.

Reply:
loadProperties() should be in both constructors. In the parameterless constructor propertyFileName is null, the loadProperties() method checks for this and if no properties file was specified, it sets the default values. The constructor with a propertyFileName parameter calls the parameterless constructor (this()) to initialize default values.
After that, it sets propertyFileName and calls loadProperties() again. This time, the method tries to load the settings from the specified properties file. If the file exists and is accessible, the settings are loaded; if not, the default values ​​are set again.
If I will delete loadProperties(), my tests will fail.

Reply:
Good poing, thank you for describing. In this case let's not calling parameterless constructor from constructor with `propertyFileName`. Instead` this();`, let's call `defaultJsonDatabasePath = getFilePath(DEFAULT_DATABASE_STORAGE_PATH);`